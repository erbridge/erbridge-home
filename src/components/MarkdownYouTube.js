import PropTypes from 'prop-types';
import React, { Component } from 'react';

import './MarkdownYouTube.scss';

class MarkdownYouTube extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    videoId: PropTypes.string.isRequired,
  };

  render() {
    const { title, videoId } = this.props;

    return (
      <p className="MarkdownYouTube">
        <iframe
          className="MarkdownYouTube__content"
          title={title}
          src={`https://www.youtube.com/embed/${videoId}`}
          frameBorder="0"
        />
      </p>
    );
  }
}

export default MarkdownYouTube;
