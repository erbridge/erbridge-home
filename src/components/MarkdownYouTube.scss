@import '../styles/_layout.scss';

.MarkdownYouTube {
  @include aspect-ratio(MarkdownYouTube__content, 16, 9);

  width: 100%;
}
