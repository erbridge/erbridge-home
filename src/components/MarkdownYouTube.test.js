import { shallow } from 'enzyme';
import React from 'react';

import MarkdownYouTube from './MarkdownYouTube';

it('renders correctly', () => {
  expect(
    shallow(<MarkdownYouTube title="Example" videoId="test-id" />),
  ).toMatchSnapshot();
});
