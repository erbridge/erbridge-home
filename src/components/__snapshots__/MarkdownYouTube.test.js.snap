// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="MarkdownYouTube"
>
  <iframe
    className="MarkdownYouTube__content"
    frameBorder="0"
    src="https://www.youtube.com/embed/test-id"
    title="Example"
  />
</p>
`;
